src/menu_parser.py

```diff
@@ -1,28 +1,34 @@
 # -*- coding: utf-8 -*-
+# pylint:disable=too-many-lines
 
 import datetime
 import re
-import sys
 import tempfile
 import unicodedata
 from abc import ABC, abstractmethod
-from enum import Enum
+from enum import Enum, auto
 from subprocess import call  # nosec: all the inputs is fully defined
-from typing import Dict, List, Optional, Pattern, Tuple
+from typing import Dict, List, Optional, Pattern, Set, Tuple
 from warnings import warn
 
 import requests
 from lxml import html  # nosec: https://github.com/TUM-Dev/eat-api/issues/19
 
-import util
-from entities import Dish, Ingredients, Menu, Price, Prices
+from entities import Canteen, Dish, Label, Menu, Price, Prices, Week
+from utils import util
+
+
+class ParsingError(Exception):
+    pass
 
 
 class MenuParser(ABC):
     """
     Abstract menu parser class.
     """
 
+    canteens: Set[Canteen] = NotImplemented
```

Comment:
Why do you assign a `NotImplemented` here?

Reply:
I saw it as a solution to provide an abstract attribute. But now I have read that one does not need to provide a value for an abstract class. I'll change it.